### DISCUSSION TITLE
extraAtributes([]) doesnt apply classes to Form Components

### DISCUSSION THREAD

**Discussion by thethunderturner** [filamentphp/filament#10334] [Help]
```php
protected function getFormSchema(): array
    {
        return [
            Grid::make()->schema([
                Section::make('Cache')
                    ->schema([
                        Toggle::make('enable_cache')
                            ->label($this->cacheEnabled ? 'Cache Enabled' : 'Cache Disabled')
                            ->disabled()
                            ->dehydrated()
                            ->extraAttributes([
                                'class' => 'align-self-center'
                            ]),
                        //...
                    ])
            ])
        ];
    }
```
My tailwind.config.js (theme)
```js
import preset from '../../../../vendor/filament/filament/tailwind.config.preset'

export default {
    presets: [preset],
    content: [
        './app/Filament/**/*.php',
        './resources/views/filament/**/*.blade.php',
        './vendor/filament/**/*.blade.php',
    ],
}
```
The class name is in my theme. I have run `npm run build`, yet still css is not applied.

**Comment by thethunderturner** [comment]
Reason: I want the class to be applied to the parent div of the toggle
![image](https://github.com/filamentphp/filament/assets/64212185/14aae67f-5c3c-4dee-bb76-9b2cc1e7d65b)
Is there a way to do that?

I just want to center it and the label vertically in the div

**Reply by leandrocfe** [reply]
Maybe using a group as parent?
```php
Group::make([
    Toggle::make('enable_cache'),
])->extraAttributes(['class' => 'align-self-center'])
```
You could also create a view layout with `getChildComponentContainer`
https://filamentphp.com/docs/3.x/forms/layout/custom#view-components

**Reply by SimonMacIntyre** [reply]
This actually did fix the problem for me, thanks! Although, I guess the original issue should be considered a bug.

**Comment by SimonMacIntyre** [comment]
Is this a bug? I'm also experiencing `extraAttributes` being ignored on my ViewField form component, trying to figure out if I'm doing something wrong...

```php
Forms\Components\ViewField::make('test')
                    ->view('components.ui.icon')
                    ->extraAttributes([
                        'class' => 'w-6 h-6',
                    ], true)
```

I even tried adding a random other thing like `'style' => 'color: blue' but nothing shows up in the html at all regarding my attempts.

**Reply by agil3st** [reply]
@SimonMacIntyre Is this issue still persist on your end? or known as bug?
This is also happening on latest version `^4.1`. I have followed [this instruction](https://tailwindcss.com/docs/styling-with-utility-classes#using-the-important-modifier) but nothing changed

Filament read-only input:
```
TextInput::make('slug')
      ->label('Slug')
      ->required()
      ->unique('products', 'slug')
      ->readOnly()
      ->extraAttributes(['class' => 'bg-gray-200!'])
      ->extraInputAttributes(['class' => 'bg-gray-200!']),
```
Result:
![Screenshot 2025-06-02 102730](https://github.com/user-attachments/assets/c28e59f5-6598-49dd-8f05-0043aa52b5e3)
![Screenshot 2025-06-02 102714](https://github.com/user-attachments/assets/9c575b1f-410c-45ff-8d94-0aab054ba308)

Expected result:
![Screenshot 2025-06-02 102835](https://github.com/user-attachments/assets/aa55e1ba-b392-4b62-9b42-a56978b30c18)
